### ISSUE TITLE
Model helpers cannot covert the "query by model" into appropriate filter 

### ISSUE THREAD

**Issue by filippomenchini** [aws-amplify/amplify-flutter#2548]
### Description

Given this schema:
```graphql
type Blog {
    id: ID!
    title: String!
    posts: [Post!] @hasMany(indexName: "postsByBlog", fields: ["id"])
}

type Post {
    id: ID!
    blogID: ID! @index(name: "postsByBlog")
    blog: Blog! @belongsTo(fields: ["userID"])
}
```

After launching `amplify codegen models` and trying to query `Post`'s related to a `Blog` an error is returned:
```
The variables input contains a field name 'blog' that is not defined for input object type 'ModelPostFilterInput'
```

Mentioned this bug [here](https://discord.com/channels/705853757799399426/1060978083538161774).

### Categories

- [ ] Analytics
- [ ] API (REST)
- [X] API (GraphQL)
- [ ] Auth
- [ ] Authenticator
- [ ] DataStore
- [ ] Storage

### Steps to Reproduce

1. Create the following schema:
```graphql
type Blog {
    id: ID!
    title: String!
    posts: [Post!] @hasMany(indexName: "postsByBlog", fields: ["id"])
}

type Post {
    id: ID!
    blogID: ID! @index(name: "postsByBlog")
    blog: Blog! @belongsTo(fields: ["userID"])
}
```
2. Create a `Blog` instance and some `Post` instances that reference the `Blog` instance.
3. Try to query posts linked to the previously created blog:
```dart
final request =
    ModelQueries.list(Post.classType, where: Post.BLOG.eq(blogId));
final response = await Amplify.API.query(request: request).response;
final data = response.data?.items ?? <Post?>[];
```
4. You should get the following error:
```
The variables input contains a field name 'blog' that is not defined for input object type 'ModelPostFilterInput'
```

### Platforms

- [X] iOS
- [X] Android
- [ ] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Environment

```bash
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.3.10, on macOS 13.0.1 22A400 darwin-arm, locale it-IT)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc2)
[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
[✓] Android Studio (version 2021.1)
[✓] VS Code (version 1.74.2)
[✓] Connected device (2 available)
[✓] HTTP Host Availability

• No issues found!
```

### Dependencies

```bash
Dart SDK 2.18.6
Flutter SDK 3.3.10
prometeo 1.0.2+102

dependencies:
- amplify_authenticator 0.2.4 [amplify_auth_cognito amplify_core amplify_flutter async aws_common collection flutter flutter_localizations intl stream_transform]
- auto_route 5.0.4 [flutter path collection meta]
- bloc 8.1.0 [meta]
- cached_network_image 3.2.3 [flutter flutter_cache_manager octo_image cached_network_image_platform_interface cached_network_image_web]
- camera 0.9.3+1 [camera_platform_interface flutter pedantic quiver flutter_plugin_android_lifecycle]
- carousel_slider 4.2.1 [flutter]
- cupertino_icons 1.0.5
- equatable 2.0.5 [collection meta]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]
- flutter_bloc 8.1.1 [flutter bloc provider]
- flutter_displaymode 0.5.0 [flutter]
- flutter_timer_countdown 1.0.5 [flutter flutter_lints]
- modal_bottom_sheet 2.1.2 [flutter]
- pinch_zoom 1.0.0 [flutter]
- preload_page_view 0.1.6 [flutter]
- repositories 0.0.1 [flutter equatable get_it amplify_flutter amplify_auth_cognito amplify_storage_s3 amplify_api path_provider image]
- salomon_bottom_bar 3.3.1 [flutter]
- scrollable_positioned_list 0.3.5 [flutter collection]
- visibility_detector 0.3.3 [flutter]

transitive dependencies:
- amplify_api 0.6.10 [amplify_api_android amplify_api_ios amplify_core amplify_flutter aws_common collection flutter meta plugin_platform_interface]
- amplify_api_android 0.6.10 [flutter]
- amplify_api_ios 0.6.10 [amplify_core flutter]
- amplify_auth_cognito 0.6.10 [amplify_auth_cognito_android amplify_auth_cognito_ios amplify_core aws_common collection flutter meta plugin_platform_interface]
- amplify_auth_cognito_android 0.6.10 [flutter]
- amplify_auth_cognito_ios 0.6.10 [amplify_core flutter]
- amplify_core 0.6.10 [aws_common collection flutter intl json_annotation meta plugin_platform_interface uuid]
- amplify_datastore_plugin_interface 0.6.10 [amplify_core collection flutter meta]
- amplify_flutter 0.6.10 [amplify_core amplify_datastore_plugin_interface amplify_flutter_android amplify_flutter_ios aws_common collection flutter meta plugin_platform_interface]
- amplify_flutter_android 0.6.10 [flutter]
- amplify_flutter_ios 0.6.10 [amplify_core flutter]
- amplify_storage_s3 0.6.10 [amplify_storage_s3_android amplify_storage_s3_ios amplify_core aws_common flutter meta plugin_platform_interface path_provider path]
- amplify_storage_s3_android 0.6.10 [flutter]
- amplify_storage_s3_ios 0.6.10 [flutter]
- archive 3.3.5 [crypto path pointycastle]
- async 2.9.0 [collection meta]
- aws_common 0.1.1 [async collection http meta stream_transform uuid]
- cached_network_image_platform_interface 2.0.0 [flutter flutter_cache_manager]
- cached_network_image_web 1.0.2 [flutter flutter_cache_manager cached_network_image_platform_interface]
- camera_platform_interface 2.3.2 [cross_file flutter plugin_platform_interface stream_transform]
- characters 1.2.1
- clock 1.1.1
- collection 1.16.0
- convert 3.1.1 [typed_data]
- cross_file 0.3.3+2 [js meta]
- crypto 3.0.2 [typed_data]
- ffi 2.0.1
- file 6.1.4 [meta path]
- flutter_blurhash 0.7.0 [flutter]
- flutter_cache_manager 3.3.0 [clock collection file flutter http path path_provider pedantic rxdart sqflite uuid]
- flutter_lints 2.0.1 [lints]
- flutter_localizations 0.0.0 [flutter intl characters clock collection material_color_utilities meta path vector_math]
- flutter_plugin_android_lifecycle 2.0.7 [flutter]
- get_it 7.2.0 [async collection]
- http 0.13.5 [async http_parser meta path]
- http_parser 4.0.2 [collection source_span string_scanner typed_data]
- image 4.0.4 [archive meta xml]
- intl 0.17.0 [clock path]
- js 0.6.5 [meta]
- json_annotation 4.7.0 [meta]
- lints 2.0.1
- matcher 0.12.12 [stack_trace]
- material_color_utilities 0.1.5
- meta 1.8.0
- nested 1.0.0 [flutter]
- octo_image 1.0.2 [flutter flutter_blurhash]
- path 1.8.2
- path_provider 2.0.11 [flutter path_provider_android path_provider_ios path_provider_linux path_provider_macos path_provider_platform_interface path_provider_windows]
- path_provider_android 2.0.22 [flutter path_provider_platform_interface]
- path_provider_ios 2.0.11 [flutter path_provider_platform_interface]
- path_provider_linux 2.1.7 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_macos 2.0.6 [flutter path_provider_platform_interface]
- path_provider_platform_interface 2.0.5 [flutter platform plugin_platform_interface]
- path_provider_windows 2.1.3 [ffi flutter path path_provider_platform_interface win32]
- pedantic 1.11.1
- petitparser 5.1.0 [meta]
- platform 3.1.0
- plugin_platform_interface 2.1.3 [meta]
- pointycastle 3.6.2 [collection convert js]
- process 4.2.4 [file path platform]
- provider 6.0.5 [collection flutter nested]
- quiver 3.2.1 [matcher]
- rxdart 0.27.7
- sky_engine 0.0.99
- source_span 1.9.0 [collection path term_glyph]
- sqflite 2.2.2 [flutter sqflite_common path]
- sqflite_common 2.4.0+2 [synchronized path meta]
- stack_trace 1.10.0 [path]
- stream_transform 2.1.0
- string_scanner 1.1.1 [source_span]
- synchronized 3.0.0+3
- term_glyph 1.2.1
- typed_data 1.3.1 [collection]
- uuid 3.0.6 [crypto]
- vector_math 2.1.2
- win32 3.1.3 [ffi]
- xdg_directories 0.2.0+2 [meta path process]
- xml 6.1.0 [collection meta petitparser]
```

### Device

iPhone 14 Simulator

### OS

iOS 16.2

### Deployment Method

Amplify CLI + Custom Pipeline

### CLI Version

10.6.1

### Amplify Config

N/A

**Comment by ragingsquirrel3** [comment]
Thanks for raising this issue @filippomenchini . Do you mind attaching a copy of your Post.dart codegen file? I suspect this is related to new updates with the CLI 10.6.x where there are some changes to codegen output related to custom primary keys (even if your backend does not use custom primary keys) that causes some issues with model helpers. I have seen codegen output like this cause issues (in my version of Post.dart):

```dart
    modelSchemaDefinition.addField(ModelFieldDefinition.belongsTo(
      key: Post.BLOG,
      isRequired: false,
      // targetName: 'blogID', used to be in codegen and model helpers relies on this
      targetNames: ['blogID'], // now you have this instead, which requires a fix in model helpers to find
      ofModelName: 'Blog'
    ));
```

I'm getting a fix ready for that issue but I want to make sure this is the same issue you are experiencing.

**Comment by filippomenchini** [comment]
Hi @ragingsquirrel3 . I posted the "Blog to many Posts" schema for easy understanding.
In my project I have this schema:
```graphql
type Tale @model @auth(rules: [
    { allow: public, provider: iam, operations: [read] },
    { allow: private, operations: [read] },
    { allow: owner, ownerField: "userID" }
]) {
    id: ID! @primaryKey
    title: String!
    thumbnail: S3Object!
    ttl: AWSTimestamp!
    user: User! @belongsTo(fields: ["userID"])
    userID: ID! @index(name: "talesByUser", sortKeyFields: ["ttl"])
    elements: [TaleElement!]! @hasMany(indexName: "taleElementsByTale", fields: ["id"])
    invitations: [Invitation!] @hasMany(indexName: "invitationsByTale", fields: ["id"])
    reactions: [Reaction!] @hasMany(indexName: "reactionsByTale", fields: ["id"])
    updatedAt: AWSDateTime!
    createdAt: AWSDateTime!
}

type TaleElement @model @auth(rules: [
    { allow: public, provider: iam, operations: [read] },
    { allow: private, operations: [read] },
    { allow: owner, ownerField: "userID" }
]) {
    id: ID! @primaryKey
    content: S3Object!
    tale: Tale! @belongsTo(fields: ["taleID"])
    taleID: ID! @index(name: "taleElementsByTale", sortKeyFields: ["createdAt"])
    user: User! @belongsTo(fields: ["userID"])
    userID: ID! @index(name: "taleElementsByUser", sortKeyFields: ["createdAt"])
    reactions: [Reaction!] @hasMany(indexName: "reactionsByTaleElement", fields: ["id"])
    updatedAt: AWSDateTime!
    createdAt: AWSDateTime!
}
```
View my `TaleElement` as a `Post` and the `Tale` as a `Blog`.

The `codegen` result is the same as yours for my `TaleElement`:
```dart
 modelSchemaDefinition.addField(ModelFieldDefinition.belongsTo(
      key: TaleElement.TALE,
      isRequired: true,
      targetNames: ['taleID'],
      ofModelName: 'Tale'
    ));
```
So, I think this is the same issue you mentioned.
Let me know if you need more info.

**Comment by ragingsquirrel3** [comment]
This has been fixed with v0.6.11. Please upgrade and let us know if you still face this issue. We will work on getting this fix into dev preview as well.

**Comment by filippomenchini** [comment]
Works as expected!
Thank you
